src/query/encoding.ts

```diff
@@ -48,3 +48,22 @@ export function isMeasure(encQ: EncodingQuery) {
   return (encQ.type === Type.QUANTITATIVE && !encQ.bin) ||
       (encQ.type === Type.TEMPORAL && !encQ.timeUnit);
 }
+
+export function scaleType(scaleType: ScaleType, timeUnit: TimeUnit, type: Type) {
+  if (scaleType !== undefined) {
+    return scaleType;
+  }
+
+  if (type === Type.QUANTITATIVE) {
+    return ScaleType.LINEAR;
+  } else if (type === Type.ORDINAL || type === Type.NOMINAL) {
+    return ScaleType.ORDINAL;
+  } else {
+    // Down here, type is guaranteeed to be Type.TEMPORAL
```

Comment:
> type is guaranteeed to be Type.TEMPORAL

How do you assume this?
  
This is not true.  Type can be enumSpec in this case.  Not to mention that we might support more type in the future.   

Check for temporal, check for enum spec and throw error for else case.